### PR TITLE
feat: infer types from field initializers

### PR DESCRIPTION
**Pull request by bennypowers**
See #12

### CODE REVIEW

**Comment by bennypowers** [comment]
As this is still subject to review, I didn't update the fixtures yet, but I _did_ commit the test output for your perusal

File: packages/analyzer/src/features/analyse-phase/creators/createClassField.js [hunk 1]
```diff
@@ -19,8 +19,16 @@ export function createField(node) {
   /**
    * Add TS type
    * @example class Foo { bar: string = ''; }
-   */ 
-  if(node.type) {
+   */
```
Comment (thepassle) [inline packages/analyzer/src/features/analyse-phase/creators/createClassField.js:22]: Lets extract this into a separate handler (see `handlers.js`) that we can reuse in multiple places, something like `handleTypeInference(fieldTemplate, node)`:

```js
  switch(node?.initializer?.kind) {
    case ts.SyntaxKind.TrueKeyword:
    case ts.SyntaxKind.FalseKeyword:
      doc.type = { text: "boolean" }
      break;
    case ts.SyntaxKind.StringLiteral:
      doc.type = { text: "string" }
      break;
    case ts.SyntaxKind.NumericLiteral:
      doc.type = { text: "number" }
      break;
    case ts.SyntaxKind.NullKeyword:
      doc.type = { text: "null" }
      break;
    case ts.SyntaxKind.ArrayLiteralExpression:
      doc.type = { text: "array" }
      break;
    case ts.SyntaxKind.ObjectLiteralExpression:
      doc.type = { text: "object" }
      break;
  }
```

Comment (bennypowers) [reply packages/analyzer/src/features/analyse-phase/creators/createClassField.js]: PTAL

File: packages/analyzer/src/utils/ast-helpers.js [hunk 2]
```diff
@@ -99,3 +99,18 @@ export const getReturnValue = returnStatement => (
  */
 export const isStaticMember = member =>
   member?.modifiers?.some?.(x => x.kind === ts.SyntaxKind.StaticKeyword);
+
+export function isBoolean(node) {
+  const v = node.initializer?.getText?.();
+  return v === 'true' || v === 'false';
+}
+
+export function isAsConst(initializer) {
```
[the remark this answers is not in the document]
Comment (bennypowers) [reply packages/analyzer/src/utils/ast-helpers.js:108]: i checked before adding these functions

`as const` _is_ the type

https://www.typescriptlang.org/play?#code/MYGwhgzhAECCEGED2A7CAXaBvAUNaY0AvNAORikEzCoYDcOAvjjqJDAMroBOAligHMAKgE8ADgFNseAsTIUmLGmkyRkKuSgkB3OIlroAFAEoGyjNAx9BoyZp3Qu14eIkmGONQYB0hEqQAjUhwrfhdJXzlA0jogA

[the remark this answers is not in the document]
Comment (bennypowers) [reply packages/analyzer/src/utils/ast-helpers.js:108]: oh sorry i misunderstood the question, I read it as "does ts api have other isFoo`

`as const` is a special case, I'm not aware of any other keyword assertions

**Comment by thepassle** [comment]
closed in favor of https://github.com/open-wc/custom-elements-manifest/pull/53

I added you as co-author to that PR

We still need to adress `as const`, but lets do that separately now

**Pull request closed without merging**